llvm-bazel/llvm-project-overlay/llvm/BUILD

```diff
@@ -30,2213 +14,1433 @@ exports_files(["LICENSE.TXT"])
-        "lib/Target/ARM/Disassembler/*.inc",
-    ]),
-    hdrs = glob([
-        "include/llvm/Target/ARM/Disassembler/*.h",
-        "include/llvm/Target/ARM/Disassembler/*.def",
-        "include/llvm/Target/ARM/Disassembler/*.inc",
-        "lib/Target/ARM/Disassembler/*.h",
+        "lib/Demangle/*.cpp",
+        "lib/Demangle/*.h",
     ]),
-    copts = llvm_copts + ["-Iexternal/llvm-project/llvm/lib/Target/ARM"],
-    deps = [
-        ":ARMDesc",
-        ":ARMInfo",
-        ":ARMUtils",
-        ":MCDisassembler",
-        ":Support",
-        ":config",
-    ],
+    hdrs = glob(["include/llvm/Demangle/*.h"]),
+    copts = llvm_copts,
+    deps = [":config"],
 )
 
-cc_library(
-    name = "ARMInfo",
-    srcs = glob([
-        "lib/Target/ARM/TargetInfo/*.c",
-        "lib/Target/ARM/TargetInfo/*.cpp",
-        "lib/Target/ARM/TargetInfo/*.inc",
-        "lib/Target/ARM/MCTargetDesc/*.h",
-    ]),
-    hdrs = glob([
-        "include/llvm/Target/ARM/TargetInfo/*.h",
-        "include/llvm/Target/ARM/TargetInfo/*.def",
-        "include/llvm/Target/ARM/TargetInfo/*.inc",
-        "lib/Target/ARM/TargetInfo/*.h",
-    ]),
-    copts = llvm_copts + ["-Iexternal/llvm-project/llvm/lib/Target/ARM"],
-    deps = [
-        ":ARMCommonTableGen",
-        ":Support",
-        ":Target",
-        ":config",
-    ],
+genrule(
+    name = "generate_vcs_revision",
+    outs = ["include/llvm/Support/VCSRevision.h"],
+    cmd = "echo '#define LLVM_REVISION \"git\"' >> $@\n" +
+          "echo '#undef LLVM_REPOSITORY' >> $@\n",
+)
+
+genrule(
+    name = "generate_static_extension_registry",
+    outs = ["include/llvm/Support/Extension.def"],
+    cmd = "echo -e '// extension handlers' >> $@\n" +
+          "echo -e '#undef HANDLE_EXTENSION' >> $@\n",
 )
 
 cc_library(
-    name = "ARMUtils",
+    name = "Support",
     srcs = glob([
-        "lib/Target/ARM/Utils/*.c",
-        "lib/Target/ARM/Utils/*.cpp",
-        "lib/Target/ARM/Utils/*.inc",
-        "lib/Target/ARM/MCTargetDesc/*.h",
+        "lib/Support/*.c",
+        "lib/Support/*.cpp",
+        "lib/Support/*.h",
+        "lib/Support/*.inc",
+        "lib/Support/Unix/*.h",
+        "lib/Support/Unix/*.inc",
     ]),
     hdrs = glob([
-        "include/llvm/Target/ARM/Utils/*.h",
-        "include/llvm/Target/ARM/Utils/*.def",
-        "include/llvm/Target/ARM/Utils/*.inc",
-        "lib/Target/ARM/Utils/*.h",
+        "include/llvm/Support/*.h",
+        "include/llvm/ADT/*.h",
+    ]) + [
+        "include/llvm/Support/VCSRevision.h",
+        "include/llvm/Support/Extension.def",
+        "include/llvm-c/Core.h",
+        "include/llvm-c/DataTypes.h",
+        "include/llvm-c/DisassemblerTypes.h",
+        "include/llvm-c/Error.h",
+        "include/llvm-c/ExternC.h",
+        "include/llvm-c/ErrorHandling.h",
+        "include/llvm-c/Support.h",
+        "include/llvm-c/Types.h",
+    ],
+    copts = llvm_copts,
+    includes = ["include"],
+    linkopts = [
+        "-lz",
```

Comment:
I now get errors compiling IREE with these build files (was clean before), even after restoring the zlib dependency:

> ERROR: [snip]/.cache/bazel/_bazel_gcmn/c17d50ae90e3b490378de262a2ee1611/external/llvm-project/mlir/BUILD:3144:10: Linking of rule '@llvm-project//mlir:mlir-tblgen' failed (Exit 1)
/usr/bin/ld.gold: error: cannot find -lz
/usr/bin/ld.gold: error: cannot find -ltinfo
bazel-out/host/bin/external/llvm-project/llvm/_objs/Support/Process.o:Process.cpp:function llvm::sys::Process::FileDescriptorHasColors(int): error: undefined reference to 'setupterm'
bazel-out/host/bin/external/llvm-project/llvm/_objs/Support/Process.o:Process.cpp:function llvm::sys::Process::FileDescriptorHasColors(int): error: undefined reference to 'tigetnum'
bazel-out/host/bin/external/llvm-project/llvm/_objs/Support/Process.o:Process.cpp:function llvm::sys::Process::FileDescriptorHasColors(int): error: undefined reference to 'set_curterm'
bazel-out/host/bin/external/llvm-project/llvm/_objs/Support/Process.o:Process.cpp:function llvm::sys::Process::FileDescriptorHasColors(int): error: undefined reference to 'del_curterm'

Reply:
Here are my patches to make IREE build, a property I'm interested in preserving ;-P

https://github.com/google/llvm-bazel/commit/2af8cba83fb593ddb76d4a2c1ac496e4d11f5ee5

Reply:
I still need an additional patch to enable dlopen, which we use

https://github.com/google/llvm-bazel/compare/chandlerc-hand-port...iree-patches-to-hand-port-2

Reply:
Ah, yes. That seems like a good change. Suggest it in this PR? Or merge that one?

Reply:
Done.